### PR TITLE
refactors for yeoman-generator 0.18.7

### PR DESCRIPTION
**Pull request by Toddses**
Okay here is the refactor. I bumped the version to 2.0.0 because it'll be a major update. There's a few smaller bug fixes in here too.

The new filesystem works a bit different so it required some tweaking of the order of functions and such. But as of this PR the user experience is pretty much the same as the current master branch. I'd like to get the remaining packages up to their latest versions as well, but I think this PR is ready for human consumption. I'd like to get it verified that we won't see the max call stack error. I have not been able to reproduce it at all with this version.

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -150,12 +151,12 @@ Generator.prototype.ohTellMeWhatYouWantWhatYouReallyReallyWant = function() {
 Generator.prototype.justIgnoreMe = function() {
 	if (this.conf.get('git')) {
 		this.logger.verbose('Copying .gitignore file');
-		this.copy('gitignore.tmpl', '.gitignore');
-		this.logger.verbose('Done copying .gitignore file');
+		this.fs.copyTpl(this.templatePath('gitignore.tmpl'), this.destinationPath('.gitignore'), { conf: this.conf });
+		//this.logger.verbose('Done copying .gitignore file');
```
Comment (wesleytodd) [inline app/index.js]: Can you clean up the comments?  For these log ones we should just uncomment them.

File: app/index.js [hunk 2]
```diff
@@ -168,14 +169,16 @@ Generator.prototype.gitIsTheShit = function() {
...
-			git.add('.', function(err) {
-				if (err) me.logger.error(err);
-			}).commit('Initial Commit', function(err, d) {
-				if (err) me.logger.error(err);
+			done();
+			//git.add('.', function(err) {
+				//if (err) me.logger.error(err);
+
+			//}).commit('Initial WP Commit', function(err, d) {
+				//if (err) me.logger.error(err);
```
Comment (wesleytodd) [inline app/index.js]: What is this commented out for?  Testing perhaps?  Doesn't seem like this was re-implemented in another way....?

File: app/index.js [hunk 3]
```diff
@@ -237,17 +242,18 @@ Generator.prototype.somethingsDifferent = function() {
 
 	if (this.conf.get('submodule') || this.conf.get('customDirs')) {
 
-		var me = this,
-			done = this.async();
+		var me = this;
+			//done = this.async();
```
Comment (wesleytodd) [inline app/index.js]: Was `this.async` changes in the new version?

File: app/index.js [hunk 4]
```diff
@@ -354,20 +367,50 @@ Generator.prototype.commitThisToMemory = function() {
...
 			//wordpress.activateTheme(me.conf.get(), done);
 			me.logger.verbose('Theme install complete');
+			//done();
+		});
+	}
+
+};
+
+// Write all the newly downloaded theme files to disk
+Generator.prototype.gandalfGotThemSkillzTho = function() {
```
Comment (wesleytodd) [inline app/index.js]: Lol, nice!

File: util/wordpress.js [hunk 5]
```diff
@@ -247,7 +247,7 @@ function installTheme(generator, config, done) {
 
 };
 
-function setupTheme(generator, config, done) {
+function setupTheme(generator, config, cb) {
```
Comment (wesleytodd) [inline util/wordpress.js]: While I am totally fine with this change, if we want this then we should probably make it to the entire code base, just for consistency.

File: app/templates/local-config.php.tmpl [hunk 6]
```diff
@@ -5,4 +5,4 @@ define('DB_USER', '<%= conf.get('dbUser') %>');
 define('DB_PASSWORD', '<%= conf.get('dbPass') %>');
 define('DB_HOST', '<%= conf.get('dbHost') %>');
 $table_prefix  = '<%= conf.get('tablePrefix') %>';
-define('WP_DEBUG', false);
+define('WP_DEBUG', true);
```
Comment (wesleytodd) [inline app/templates/local-config.php.tmpl]: This should be changed back.

File: app/templates/gitignore.tmpl [hunk 7]
```diff
@@ -8,7 +11,7 @@ node_modules
 
 # Ignore config
 local-config.php
-<% if (conf.get('ignoreWPCore')) { %>
+<% if (conf.get('ignoreWPConfig')) { %>
 wp-config.php
```
Comment (wesleytodd) [inline app/templates/gitignore.tmpl]: Do you think you could pull this change into another PR based on the existing master?  I would like to get this as a patch on the 1.4.X version as well.

**Comment by wesleytodd** [comment]
I especially like your continuation of the random function names :+1:

**Comment by wesleytodd** [comment]
Once you get those changes made I will pull it down and go through it a few times to make sure I don't get any issues, but other than that it all looks great!  Thanks for the hard work!

File: app/index.js (hunk 1 above)
Comment (Toddses) [inline app/index.js]: Yes I will. I commented out stuff and forgot to go back and clean those out.

Comment (wesleytodd) [inline app/index.js]: No worries, that is what peer review is for!!

File: app/index.js (hunk 2 above)
Comment (Toddses) [inline app/index.js]: Well the new fs only writes files at the end of execution. So trying to add and commit files with git breaks when there's no files to add or commit. I tried out doing a lot of small `fs.commit()` calls through each step but ultimately decided it might be more efficient to break those out into bigger chunks. These comments just need to be cleaned up. The initial commit is done after the first `fs.commit()`, after wordpress is installed.

Perhaps it would be better, and closer to the original, to go ahead and `fs.commit()` the gitignore and do an initial commit with just that file.

File: app/index.js (hunk 3 above)
Comment (Toddses) [inline app/index.js]: no, clean up needed. with the new fs, it seemed like this function didn't need to run asynchronously any more.

File: util/wordpress.js (hunk 5 above)
Comment (Toddses) [inline util/wordpress.js]: will do. i couldn't quite get it to work the same using `async()` in this way. hence the change. i may do a little more experimenting and see if we can avoid the change.

File: app/templates/gitignore.tmpl (hunk 7 above)
Comment (Toddses) [inline app/templates/gitignore.tmpl]: np

File: app/templates/local-config.php.tmpl (hunk 6 above)
Comment (Toddses) [inline app/templates/local-config.php.tmpl]: np

File: app/index.js (hunk 2 above)
Comment (Toddses) [inline app/index.js]: That's what I'll do here, go ahead and make an initial commit with just that gitignore, as that's more in line with v1.4.x

Comment (wesleytodd) [inline app/index.js]: Since we are doing 2.0, changing the behavior is just fine.  I must have missed the commit later down.  You should leave the new behavior, I like it being all done at once later.

File: app/index.js (hunk 3 above)
Comment (wesleytodd) [inline app/index.js]: If it is not necessary then it can go, I was just checking.

File: app/index.js (hunk 2 above)
Comment (Toddses) [inline app/index.js]: :+1:

**Comment by Toddses** [comment]
OK I've made the changes.

**Comment by wesleytodd** [comment]
Ok, merge this with only a few changes.  I am guessing you didn't test the vagrant stuff, but I fixed the issues there.  Also, there was a small issue where the theme files were not committed before trying to run the build steps.  It was just a missing call to `this.async`, so I fixed that as well.  While I was at it I also updated some things like using `bind` instead of `var me = this`.

So, If you could pull down develop and do a last round of testing after my changes that would be awesome.  If it fully works for you then I will release 2.0!!

**Comment by Toddses** [comment]
OK I ran through it a bunch of times with no issues. Looks like its good to go!

**Comment by wesleytodd** [comment]
Same here, I merged your other one as well, so we should be good to go.  Ill merge it up to master and release later tonight.
